### ISSUE TITLE
Add more explanation on how loss function and optimization works

### ISSUE THREAD

**Issue by svenvanderburg** [carpentries-incubator/deep-learning-intro#251]
It would be nice to add a bit more theory on loss functions and optimization. We often get questions about it and explain along the way by improvising, but it would be nice if there is some material about it with pictures. Something like: 
![image](https://user-images.githubusercontent.com/9945255/195030822-290b941b-b93d-4689-b1d7-75a5a70cbfef.png)
and
![image](https://user-images.githubusercontent.com/9945255/195030654-97046402-54c7-4c80-98be-238416449ce4.png)

This would fit in https://carpentries-incubator.github.io/deep-learning-intro/02-keras/index.html#5-choose-a-loss-function-and-optimizer

**Comment by svenvanderburg** [comment]
# Sven's proposal to solve this and #106 

## Introduce loss function in the introduction (episode 1)
In episode 1 we add the objective 'Describe what a loss function is'. We will test this by adding a short exercise 'Compute the Mean Absolute Error loss' on a small set of predictions and true values (n=5).
We will make it clear that the goal for machine learning is to minimise the loss, this is called 'optimization' and we will get back to how that works in episode 3.

## Explain optimization (with gradient descent), learning rate and batch size in episode 3
The first time that we have to pick an optimizer is here: https://carpentries-incubator.github.io/deep-learning-intro/02-keras/index.html#5-choose-a-loss-function-and-optimizer . However in episode 2 I think we just want to train our first model as soon as possible, so I don't want to introduce optimization just yet. It is good to mention though that it will be explained in episode 3.

I suggest to explain optimization, learning rate and batch size here: https://carpentries-incubator.github.io/deep-learning-intro/03-monitor-the-model/index.html#5-choose-a-loss-function-and-optimizer . Right before we have to pick the optimization algorithm.

I suggest to add the objectives: 'Explain how neural networks learn' and 'Understand how batch size and learning rate can affect the training progress'. We will test this:

### Exercise
What is the goal of optimization?
A. To find the weights that maximize the loss function
B. To find the weights that minimize the loss function

What happens in one gradient descent step?
A. The weights are adjusted so that we move down the slope of the loss function
B. The weights are adjusted so that we move up the slope of the loss function

When the learning rate is too low:
(multiple answers might apply)
A. The learning requires too many update steps before reaching the minimum point, so learning is very slow.
B. The learner swiftly reaches the minimum point and stays there
C. This causes drastic updates that leads to divergent behaviours moving away from the minimum point
D. A local minimum is reached, and the learning steps are not big enough to move out of it.
![image](https://github.com/carpentries-incubator/deep-learning-intro/assets/9945255/f5054477-3c92-4888-ae17-a94f4837a3ac)

When the batch size is increased:
(multiple answers might apply)
A. The number of samples in an epoch also increases
B. The number of batches in an epoch goes down
C. The training progress is less jumpy, because more samples are consulted in each training step
D. The memory load of the training process is increased
